app/assets/stylesheets/components/header.scss

```diff
@@ -101,18 +103,23 @@
     opacity: var(--dropdown-opacity, 0);
     visibility: var(--dropdown-visibility, hidden);
   }
-
   &.showing,
-  &.desktop:hover,
-  &.desktop:focus {
+  &.desktop:hover {
     --dropdown-transform: translateY(0) rotateX(0);
     --dropdown-delay: 0;
     --dropdown-opacity: 1;
     --dropdown-visibility: visible;
 
     .crayons-header__menu__trigger {
+      border: 2px solid var(--header-bg-hover);
       color: var(--header-icons-color-hover);
-      background: var(--header-bg-hover);
+    }
+  }
+  &.showing .focus-visible:focus,
+  .focus-visible:focus {
+    &.crayons-header__menu__trigger {
```

Comment:
I think we can optimize this entire section and make it more readable by relying on CSS variables and `:focus-within`:

```
.crayons-header__menu {
  --trigger-color: var(--header-icons-color);
  --trigger-border-color: transparent;
  ...

  &__trigger {
    ...
    color: var(--trigger-color);
    border: 2px solid var(--trigger-border-color);
  }

  &.showing,
  &.desktop:hover {
    --trigger-border-color: var(--header-bg-hover);
    --trigger-color: var(--header-icons-color-hover);
    ...
  }
  &:focus-within,
  &.showing,
  &.desktop:hover:focus-within {
    --trigger-border-color: var(--header-button-focus-color);
    --trigger-color: var(--header-button-focus-color);
  }
}
```

^ it also requires [removing this piece of code](https://github.com/forem/forem/blob/master/app/assets/stylesheets/tag-edit.scss#L32-L34) - which, to be fair, shouldn't be in that file anyway. and it seems like causing more confusion (especially with specificity) anyway.

Thoughts?

Reply:
Actually... I've just pushed this code :D Let me know if it makes sense.

Reply:
The variables definitely making things more readable 👍  Only issue I see though is that removing use of `focus-visible` means that the focus border appears around the profile image on click as well as on keyboard focus. It also seems to be hanging around even if you click the menu and then start tabbing and moving focus away.

I'll tweak what you've added and then hopefully we should be good

Reply:
I thought that's what was originally suggested in this PR, no? Nevertheless, feel free to make any adjustments you want :)

Reply:
It's been a long thread 😅  The proposal is to show the border around the profile image when it's focused - with the changes it's persisting after a mouse click even though the focus has moved elsewhere 🙃

Reply:
> It's been a long thread

Indeed. I no longer know what's good or bad :D As I said - adjust it the way you think makes the most sense 👍